ansible_collections/arista/avd/plugins/plugin_utils/eos_validate_state_utils/avdtestbase.py

```diff
@@ -43,11 +46,37 @@ def render(self) -> dict:
         """
         return getattr(self, "test_definition", None) or {}
 
+    def get_host_structured_config(self, host: str) -> dict:
+        """
+        Retrieves and returns the structured configuration for a specified host.
+
+        The function fetches the structured_config from hostvars. It ensures the returned object is a
+        dictionary, converting Ansible dictionary-like objects to a standard dictionary if necessary.
+
+        Args:
+            host (str): Hostname to retrieve the structured_config.
+
+        Returns:
+            dict: Structured configuration for the host.
+        """
+        if host not in self.hostvars:
+            raise AristaAvdError(f"Host '{host}' is missing from the hostvars.")
+        struct_cfg = get(self.hostvars, host, separator="..")
+
+        # Check if struct_cfg is a mapping object (e.g. Ansible 'hostvars' object or regular dict)
+        if not isinstance(struct_cfg, Mapping):
+            raise AristaAvdError(f"Host '{host}' structured_config is not a dictionary or dictionary-like object.")
+
+        if not isinstance(struct_cfg, dict):
```

Comment:
Why do we need this? I could be using chain-maps to save memory for inherited data.
Also running dict() on hostvars seems unnessary. I understand you are nitpicking here to convert HostVarsVars to a dict, but that is not free at all. It will perform some templating of all keys, even if we only need a few.